PW/src/mod_sirius.f90

```diff
@@ -480,6 +480,9 @@ SUBROUTINE calc_rhoc_radial_integrals( iat, nq, q, rhoc_ri ) BIND(C)
     !
     REAL(DP) :: aux(rgrid(iat)%mesh), r
     INTEGER :: iq, ir, nr
+    IF (.NOT. allocated( upf(iat)%rho_atc )) THEN
```

Comment:
ok, I see, this is to zero the radial integral in case of no nlcc (non linear core corrections) for that specific atom (`iat`).

Given that `upf%rho_atc` is allocated `IF ( upf%nlcc )` ([source](https://github.com/electronic-structure/q-e-sirius/blob/merge_qe_73/Modules/read_pseudo.f90#L322-L326)), this should be equivalent since the result is indeed initialized to zero [here](https://github.com/electronic-structure/SIRIUS/blob/develop/src/radial/radial_integrals.hpp#L258)

Reply:
I still think rhoc_ri = 0.D0 is needed even if it is zeroed on sirius side. `rhoc_ri` is an output variable, so you can't rely on external code to set it to zero.